src/ui/ui_i18n/README.md

```diff
@@ -0,0 +1,169 @@
+## I18n engine
+
+[React-intl](https://github.com/yahoo/react-intl) is built around
+[intl-messageformat](https://github.com/yahoo/intl-messageformat),
+so both react and angular frameworks will use the same engine and
+the same message syntax. The engine uses the ICU Message syntax
+and works for all CLDR languages which have pluralization rules
+defined. `intl-messageformat` package exposes `IntlMessageFormat`
+constructor. Messages are provided into the constructor as a
+string message, or a pre-parsed AST object.
+
+```
+import IntlMessageFormat from 'intl-messageformat';
+
+const msg = new IntlMessageFormat(message, locales, [formats]);
+```
+
+The string `message` is parsed, then stored internally in a
+compiled form that is optimized for the `format()` method to
+produce the formatted string for displaying to the user.
+
+```
+const output = msg.format(values);
+```
+
+Creating instances of `IntlMessageFormat` is an expensive operation.
+[Intl-format-cache](https://github.com/yahoo/intl-format-cache)
+library is simply to make it easier to create a cache of format
+instances of a particular type to aid in their reuse. Under the
+hood, this package creates a cache key based on the arguments passed
+to the memoized constructor.
+
+```
+import memoizeIntlConstructor from 'intl-format-cache';
+
+const getMessageFormat = memoizeIntlConstructor(IntlMessageFormat);
+```
+
+`Intl-messageformat` package assumes that the
+[Intl](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)
+global object exists in the runtime. `Intl` is present in all modern
+browsers and Node.js > 0.10. In order to load `Intl MessageFormat`
+in Node.js we should simply `require()` this package (in Node.js,
+the data for all 200+ languages is loaded along with the library):
```

Comment:
The data for all languages is loaded only for Node.js environment, but it will be ignored when bundling for the browser with Webpack (package.json has 2 fields: "main": "index.js" for Node.js and "jsnext:main": "src/main.js" for Webpack).